src/components/builder.ts

```diff
@@ -490,7 +490,8 @@ export class Builder {
                 if (!step.args) {
                     step.args = []
                 }
-                if (this.isMiktex && ((step.command === 'latexmk' && !step.args.includes('-lualatex') && !step.args.includes('-pdflua')) || step.command === 'pdflatex')) {
+                const isLuaLatex = step.args.includes('-lualatex') || step.args.includes('-pdflua') || step.args.includes('--lualatex') || step.args.includes('--pdflua')
```

Comment:
It is not appropriate to add `--max-print-line=` when the option `-pdflualatex="COMMAND"` is given.  Just add `step.args.includes('-pdflualatex') || step.args.includes('--pdflualatex')` to the alternation.